llvm/lib/Target/AMDGPU/VOP1Instructions.td

```diff
@@ -222,6 +222,13 @@ def VOPProfile_MOV : VOPProfile <[i32, i32, untyped, untyped]> {
   let InsVOPDXDeferred = (ins VSrc_f32_Deferred:$src0X);
   let InsVOPDY = (ins Src0RC32:$src0Y);
   let InsVOPDYDeferred = (ins VSrc_f32_Deferred:$src0Y);
+
+  let HasModifiers = 1;
+  let HasClamp = 1;
+
```

Comment:
I think the behavior across targets is fragmented. GFX9 and earlier = no input or output mods. GFX10 = input mods, no output mods. GFX11+ = input and output mods. The real way to support that behavior is to have different version of the V_MOV pseudo on different subtargets, but that does not seem worth it. An option is to have one pseudo with input mods only, but that will still not be correct for gfx9 and earlier. Do you see any way around that?

Reply:
I would say that there is no need for v_mov_b32 with modifiers. It can always be xor/and/or with constant or some other instruction. I can't think of any reasonable use case. We're just introducing special cases in a lot of places.

I would exclude v_mov_b32 from this patch and just focus on other instructions. Unless you know of any case where this would be valuable?

Reply:
> I would say that there is no need for v_mov_b32 with modifiers. It can always be xor/and/or with constant or some other instruction. I can't think of any reasonable use case. We're just introducing special cases in a lot of places.
> 
> I would exclude v_mov_b32 from this patch and just focus on other instructions. Unless you know of any case where this would be valuable?

I agree. Excluding v_mov_b32 from the patch because it is difficult to support in all the cases and not that useful is a good solution.